### PR TITLE
Docs: similarity search

### LINKED ISSUE #2027: [Similarity search] Add documentation guides

### PR DESCRIPTION
**Pull request by dvsrepo**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #2027

**Type of change**

Please delete options that are not relevant.

- [x] Documentation update

**How Has This Been Tested**

**Checklist**

- [ ] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [ ] follows the style guidelines of this project
- [ ] I did a self-review of my code
- [ ] I added comments to my code
- [ ] I made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by davidberenstein1957 — COMMENTED** [review]
@dvsrepo looks good to me, but we might include some other things.

- [x] further define rg.load query functionalities
- [x]  remove `fast-sentence-transformers`-> `sentence-transformers`
- [x]  add references to Lucene within the query page --> this doesn't seem to be possible to do with the query string dsl

Will not tackle in this  PR:
- [x]  add weak supervision client rules to docs 
- [x]  small tutorial using UI as well (small priority)

File: docs/_source/guides/features/semantic-search.ipynb [hunk 1]
```diff
@@ -0,0 +1,1292 @@
...
+    "It's completely up to the user which encoding or embedding mechanism to use for producing these vectors. In the \"Encode text fields\" section of this document you will find several examples and details about this process, using open source libraries (e.g., Hugging Face) as well as paid services (e.g., Cohere or OpenAI).\n",
+    "\n",
+    "Currently, Argilla uses vector search only for searching similar records (nearest neighbours) of a given vector. This can be leveraged from Argilla UI as well as the Python Client. In the future, vector search could be leveraged as well for free text queries using Argilla UI.\n",
+    "    \n",
+    "</div>\n",
+    "\n",
+    "\n",
+    "## Setup Elasticsearch or Opensearch with vector search support\n",
+    "\n",
+    "In order to use this feature you should use Elasticsearch at least version `8.5.x`or Opensearch `2.3.0`. We provide pre-configured docker-compose files in the root of Argilla's [Github repository](https://github.com/argilla-io/argilla).\n",
```
Comment (frascuchon) [inline docs/_source/guides/features/semantic-search.ipynb]: Minor supported version for OpenSearch is 2.2.0

```suggestion
    "In order to use this feature you should use Elasticsearch at least version `8.5.x`or Opensearch `2.2.0`. We provide pre-configured docker-compose files in the root of Argilla's [Github repository](https://github.com/argilla-io/argilla).\n",
```

File: docs/_source/guides/features/semantic-search.ipynb [hunk 2]
```diff
@@ -0,0 +1,1292 @@
...
+    "\n",
+    "<div class=\"alert alert-warning\">\n",
+    "\n",
+    "Warning\n",
+    "\n",
+    "If you had Argilla running with Elasticsearch 7.1.0 you need to migrate to at least version 8.5.x. Please check the section \"[Migrating from Elasticsearch 7.1.0 to 8.5.x](#migrating-from-elasticsearch-710-to-85x)\"\n",
```
Comment (frascuchon) [inline docs/_source/guides/features/semantic-search.ipynb]: The section link does not work

Comment (dvsrepo) [reply docs/_source/guides/features/semantic-search.ipynb]: Don't remember adding this link ref. Could you take a look @davidberenstein1957 ,  I'll not tackle this this week

**Comment by dvsrepo** [comment]
Only thing left to this PR after the code review is to include a few screenshots in the UI section and we can merge this. Could you include them @davidberenstein1957 ?

File: docs/_source/guides/features/semantic-search.ipynb (hunk 2 above)
Comment (davidberenstein1957) [reply docs/_source/guides/features/semantic-search.ipynb]: I resolved this by adding casing.

**Pull request merged**